projects/igniteui-angular/src/lib/carousel/carousel.component.html

```diff
@@ -21,30 +21,32 @@
     </a>
 </ng-template>
 
-
-<div *ngIf="showIndicators" [ngClass]="indicatorsOrientationClass">
+<div *ngIf="showIndicators" [ngClass]="indicatorsOrientationClass" [attr.role]="'tablist'">
     <div *ngFor="let slide of slides"
         class="igx-carousel-indicators__indicator"
         (click)="select(slide)"
+        [id]="setId(slide)"
```

Comment:
Use a getter instead of a method for the attributes. Same is applicable to `attr.aria-label`. You can host bind from within the component also with a getter.

Reply:
I agree with you that there is no need to use a method here, but I can not use a getter because the id and the labels that are constructed depends on the slide index and I can not pass a parameter to a getter, but maybe is better here to just construct the string literal directly in the template ... I don't see any benefit in using a function for that simple purpose. I can no use a host binding because tab elements are not in a separated component, they are part of the carousel component template.